docs/assets/javascripts/version-select.js

```diff
@@ -37,9 +39,51 @@ window.addEventListener("DOMContentLoaded", function() {
         var deprecatedVersionsArray = new Array();
         var versionsArray = new Array();
 
+        // Used to decorate options with a numeric sort order
+        function decorateWithSortOrder(option) {
```

Comment:
I am very confused what is this all about 😱

Reply:
Here's the problem we're solving with this function. We're sorting versions that have dots, and this messes with the correct sorting when comparing strings:
```
> console.log('2.5.123' > '10.1.456')
> true // Whoops. Version 10 should come after version 2...
```

This function takes the version number of the option, say `2.5.123`, in the form `major.minor.patch` and transforms it into a number we can then use for sorting.

It does so by multiplying the main version number by 1 million, the subversion by 1 thousand, and the patch number by 1, and then summing them all.

So `2.5.123` becomes `2500123` and `10.1.456` becomes `10100456`, fixing our sorting issue:
```
> console.log(2500123 > 10100456)
> false // Better.
```

I know...lots of lines to solve a tiny problem. If you have better ideas, I'm all ears.
I've seen other solutions, even less readable than this one 😅

Reply:
Yeah... I really can't read into this 🙈 I will approve to unblock you, but if you prefer another set of eyes, feel free to request their review